plugins/cros-ec/fu-cros-ec-firmware.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2020 Benson Leung <bleung@chromium.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-fmap-firmware.h"
+#include "fu-cros-ec-common.h"
+#include "fu-cros-ec-firmware.h"
+
+#define MAXSECTIONS	2
+
+struct _FuCrosEcFirmware {
+	FuFirmware			parent_instance;
+	struct cros_ec_version		version;
+	FuCrosEcFirmwareSection		sections[MAXSECTIONS];
+};
+
+G_DEFINE_TYPE (FuCrosEcFirmware, fu_cros_ec_firmware, FU_TYPE_FIRMWARE)
+
+static gboolean
+fu_cros_ec_firmware_parse (FuFirmware *firmware,
+			   GBytes *fw,
+			   guint64 addr_start,
+			   guint64 addr_end,
+			   FwupdInstallFlags flags,
+			   GError **error)
+{
+	FuCrosEcFirmware *self = FU_CROS_EC_FIRMWARE (firmware);
+	g_autoptr(FuFirmware) fmap_firmware = fu_fmap_firmware_new ();
```

Comment:
I expect this also goes away in the new call too.  If you need to refer the fmap firmware it should be something like `FU_FMAP_FIRMWARE ()`

Reply:
Right. If `FuCrosEcFirmware` is a subclass of `FuFmapFirmware` then you can just cast it to `FU_FIRMWARE()` without creating a new object.

Reply:
Done.